src/System.Private.CoreLib/src/System/Threading/EventWaitHandle.Windows.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.IO;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Threading
+{
+    public partial class EventWaitHandle
+    {
+        private EventWaitHandle(SafeWaitHandle handle)
+        {
+            SafeWaitHandle = handle;
+        }
+
+        private static void VerifyNameForCreate(string name)
+        {
+            if (null != name && ((int)Interop.Constants.MaxPath) < name.Length)
```

Comment:
At some point I seem to remember there being discussion about just letting the OS tell us if the name is too long, rather than trying to do our own upfront verification.  Am I remembering correctly?  Where did we land on that?